fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/SortedQuerySegmentAggregator.java

```diff
@@ -129,6 +122,15 @@ public SqlQueryData buildQuery() throws Exception {
             // Build FROM clause
             buildFromClause(sqlSortQuery, resourceType.getSimpleName());
 
+            // Gather up all bind variables from the query segments
+            // An important step here is to add _id and _lastUpdated
+            List<Object> allBindVariables = new ArrayList<>();
+            allBindVariables.addAll(idsObjects);
+            allBindVariables.addAll(lastUpdatedObjects);
+            for (SqlQueryData querySegment : this.querySegments) {
+                allBindVariables.addAll(querySegment.getBindVariables());
+            }
```

Comment:
this looks like its more related to that issue we were getting when _lastUpdate and _id weren't getting handled right.  i guess there still some case where we're failing to use those properly?  maybe we need to get this fix into 4.0.1?

Reply:
already in 4.0.1 :)